habitat_baselines/common/env_utils.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import random
+from typing import Type
+
+import habitat
+from habitat import Config, Env, VectorEnv, make_dataset
+
+
+def make_env_fn(
+    task_config: Config, rl_env_config: Config, env_class: Type, rank: int
+) -> Env:
+    r"""
+    Creates an env of type env_class with specified config and rank.
+    This is to be passed in as an argument when creating VectorEnv.
+    Args:
+        task_config: task config file for creating env.
+        rl_env_config: RL env config for creating env.
+        env_class: class type of the env to be created.
+        rank: rank of env to be created (for seeding).
+
+    Returns:
+        env object created according to specification.
+    """
+    dataset = make_dataset(
+        task_config.DATASET.TYPE, config=task_config.DATASET
+    )
+    env = env_class(
+        config_env=task_config, config_baseline=rl_env_config, dataset=dataset
+    )
+    env.seed(rank)
+    return env
+
+
+def construct_envs(config: Config, env_class: Type) -> VectorEnv:
+    r"""
+    Create VectorEnv object with specified config and env class type.
+    To allow better performance, dataset are split into small ones for
+    each individual env, grouped by scenes.
+    Args:
+        config: configs that contain num_processes as well as information
+        necessary to create individual environments.
+        env_class: class type of the envs to be created.
+
+    Returns:
+        VectorEnv object created according to specification.
+    """
+    trainer_config = config.TRAINER.RL.PPO
+    rl_env_config = config.TRAINER.RL
+    config = config.TASK_CONFIG  # exclude trainer-specific configs
```

Comment:
Usually one variable is used for one meaning, re-assign config isn't good idea, as it can have totally different meaning (experiment/baseline config and task config.) depends on code line.